### PR TITLE
LPD-35740 Cover guest cart reconciliation for B2B and B2X upon user login

### PR DESCRIPTION
**Pull request by gianmarcobrunialti**
https://liferay.atlassian.net/browse/LPD-35740

### CODE REVIEW

**Comment by gianmarcobrunialti** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/events/LoginPostAction.java [hunk 1]
```diff
@@ -104,33 +111,57 @@ private void _addDefaultAccountRoles(HttpServletRequest httpServletRequest)
...
+		if (userAccountEntries.isEmpty()) {
+			return null;
 		}
 
-		long userId = _portal.getUserId(httpServletRequest);
+		return userAccountEntries.get(0);
+	}
+
+	private AccountEntry _getPersonAccountEntry(
+			long userId, HttpServletRequest httpServletRequest)
```
Comment (smottal) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/events/LoginPostAction.java:163]: probably these should be sorted

Comment (gianmarcobrunialti) [reply modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/events/LoginPostAction.java:163]: private methods should be already sorted
```
_getAccountEntry
_getBusinessAccountEntry
_getPersonAccountEntry
```

Comment (gianmarcobrunialti) [reply modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/events/LoginPostAction.java]: Resorted params. Thx 👍🏼

**Comment by gianmarcobrunialti** [comment]
ci:test:sf

**Comment by gianmarcobrunialti** [comment]
ci:test:relevant

**Comment by gianmarcobrunialti** [comment]
ci:test:commerce-functional

**Review by smottal — APPROVED** [review]

**Comment by gianmarcobrunialti** [comment]
ci:test:relevant

**Comment by gianmarcobrunialti** [comment]
ci:test:commerce-functional

**Comment by gianmarcobrunialti** [comment]
ci:test:stable

**Pull request closed without merging**